### PR TITLE
Feature/sample sheet support

### PR DESCRIPTION
**Pull request by pavlemarinkovic**
- Added support for autogenerating sample sheet function(s) for Nextflow app wrapper.
- sbpack will now avoid overwriting previous version of sb_nextflow_schema.yaml
- Enums will be better parsed and generated
- New --auto mode tries to locate as many files and inputs for generating the app automatically
- The logic for uploading the app code package and the app itself has been updated. It now first tires to create the app wrapper, and if it succeeds then tries to upload the code package.

### CODE REVIEW

File: sbpack/lib.py [hunk 1]
```diff
@@ -71,7 +72,8 @@ def normalize_to_map(obj: Union[list, dict], key_field: str):
         raise RuntimeError("Expecting a dictionary or a list here")
 
 
-def normalize_to_list(obj: Union[list, dict], key_field: str, value_field: str):
+def normalize_to_list(
+        obj: Union[list, dict], key_field: str, value_field: Union[str, None]):
```
Comment (borislavd88) [inline sbpack/lib.py:76]: Maybe better Optional[str] instead of Union[str, None].

File: sbpack/noncwl/utils.py [hunk 2]
```diff
@@ -279,6 +305,63 @@ def get_entrypoint(path):
...
+        return None
+
+
+def get_executor_version(string):
+    result = re.findall(
+        r"\[Nextflow]\([^(]+(%E2%89%A5|%E2%89%A4|=|>|<)(\d{2}\.\d+\.\d+)[^)]+\)",
+        string
+    )
+    if result:
+        sign, version = result.pop(0)
```
Comment (borislavd88) [inline sbpack/noncwl/utils.py:341]: is unpacking here ok? 4 groups are expected in regular expression?

Comment (borislavd88) [reply sbpack/noncwl/utils.py:341]: Sorry, maybe it is ok. If this is example for executor version string: "[Nextflow]([=12.2.4])" ?

**Review by borislavd88 — APPROVED** [review]

File: sbpack/lib.py (hunk 1 above)
Comment (pavlemarinkovic) [reply sbpack/lib.py]: Changed it in the latest commit :)

**Pull request merged**